test/test_jit.py

```diff
@@ -1421,18 +1429,20 @@ def func(lim):
 
             st = second + third
             fs = first + second
-            return third, st, fs
+            zero = FIXME_zerol()
+            return third + zero, st + zero, fs + zero
 
         inputs = self._make_scalar_vars([10], torch.int64)
         self.checkScript(func, inputs, optimize=True)
 
     def test_if(self):
         def func(a, b):
+            zero = FIXME_zerol()
             d = 3
             if a > 10:
-                a = 3 + d
+                a = zero + 3 + d
             else:
-                b = 3 + d
+                b = zero + 3 + d
```

Comment:
Can we please avoid reusing existing tests, with well-defined scope for checking this extra behavior? If I were to refactor the test in the future, it's possible I would remove this stuff, because it doesn't contribute to `test_if` in any way

Reply:
The reason why I added this in here is that `a` and `b` are both treated as DynamicType. `3 + d` is a IntType. We disallow assigning DynamicType a IntType value, so I had to convert the IntType value to a DynamicType value somewhere.

Reply:
Ugh, we should probably relax those constraints. If you can't change tensors to numbers, natural things like
```python
loss = 0.
for o in outputs:
    loss += o.sum()
```
will break.

Reply:
Actually can we just fix the broken tests by removing the scalar literals (use tensor math only)? The fact that so many of them needed to be change is a sign that our current strategy is way too strict

Reply:
Do you mean changing all of the python numbers to scalar tensors, or in general just getting rid of python numbers in the tests and sticking in tensors somehow?